muspinsim_gen/muspinsim_gen.xml

```diff
@@ -0,0 +1,116 @@
+<tool id="muspinsim_gen" name="MuSpinSim Generator" version="@TOOL_VERSION@+galaxy@WRAPPER_VERSION@" python_template_version="3.5" profile="22.05" license="MIT">
+    <description>Generate MuSpinSim config from a structure file</description>
+    <macros>
+        <!-- version of underlying tool (PEP 440) -->
+        <token name="@TOOL_VERSION@">2.2.0</token>
+        <!-- version of this tool wrapper (integer) -->
+        <token name="@WRAPPER_VERSION@">0</token>
+        <!-- citation should be updated with every underlying tool version -->
+        <!-- typical fields to update are version, month, year, and doi -->
+        <token name="@TOOL_CITATION@">
+            @software{muspinsim,
+                author = {Sturniolo, Simone and Liborio, Leandro and Owen, Josh and Mudaraddi, Anish and Davies, Joel and Wilkinson, John and {Muon Spectroscopy Computational Project}},
+                license = {MIT},
+                title = {{muspinsim}},
+                url = {https://github.com/muon-spectroscopy-computational-project/muspinsim},
+                version = {v2.2.0},
+                month = {3},
+                year = {2023}
+                doi = {10.5281/zenodo.7704635}
+            }
+        </token>
+    </macros>
+    <creator>
+        <person givenName="Joel" familyName="Davies" url="https://github.com/joelvdavies" identifier="https://orcid.org/0000-0002-4153-6819"/>
+        <organization url="https://muon-spectroscopy-computational-project.github.io/index.html" name="The Muon Spectroscopy Computational Project"/>
+    </creator>
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">muspinsim</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        structure_name_internal=\$(sed 's/ /\_/g' <<< '$structure_file.name') &&
+        ln -s '$structure_file' \$structure_name_internal &&
```

Comment:
While testing, I ran into the problem that ASE will fail to read the structural input unless the name of my dataset in Galaxy has the right file extension. So if I provide `Si.cell`, everything's fine, but if my dataset is just called `Si` ASE can't load it. This was also the case for xyz, magres etc.

We run into this problem (or problems like it) a fair amount. I'd be tempted to say we should use something like `struct.$structure_file.extension` as our filepath, which will use the file type identified by Galaxy - which is reliably `cell`, `cif`, `xyz` or `extxyz`. However, it will never be `magres`, because we don't have a datatype for it yet (I tried a `Si.txt` file and ASE didn't read that). So our current options are: stick with it as is, and hope that the user always gives the dataset a name which ends with the extension, or implement a general solution that won't work with magres until we create a datatype for it.